src/dlgprefshoutcast.cpp

```diff
@@ -257,7 +293,35 @@ void DlgPrefShoutcast::slotApply()
     m_pConfig->set(ConfigKey(SHOUTCAST_PREF_KEY, "custom_artist"), ConfigValue(custom_artist->text()));
     m_pConfig->set(ConfigKey(SHOUTCAST_PREF_KEY, "custom_title"),  ConfigValue(custom_title->text()));
     m_pConfig->set(ConfigKey(SHOUTCAST_PREF_KEY, "metadata_format"), ConfigValue(metadata_format->text()));
+}
+
+void DlgPrefShoutcast::setDialogEnabled(bool enabled) {
+        comboBoxServerType->setEnabled(enabled);
```

Comment:
usually the way to do this is to find a parent widget and set its enabled state.  Then the enabled state is propagated to all the children.